### PR TITLE
refactor: legacy form setting backward compatibility handler

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

This pr simplify logic which I developed in https://github.com/impress-org/givewp/issues/4608 and add more abstraction. I created separate properties for each legacy setting and update handler to handle them internally with exposing it developer.

## Affects

- Added new properties in `Give\Form\Template`
- Added new static functions to get field configs in `Give\Form\Template\Options`
- Updated logic to store settings in `Give\Form\Template\LegacyFormSettingCompatibility`
- Update template option config of exiting templates in favor of the above changes.

## What to test

Edit and save from template setting. Review donation form view on the frontend. 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: src/Form/Template.php [hunk 1]
```diff
@@ -22,26 +21,74 @@
...
+	 * @var string $donateNowButtonLabel
+	 */
+	public $donateNowButtonLabel = '';
+
+	/**
+	 * Flag to check whether or not show donation form introduction text.
+	 *
+	 * @var string $showDonationIntroductionContent
+	 */
+	public $showDonationIntroductionContent = 'disabled';
```
Comment (JasonTheAdams) [inline src/Form/Template.php:77]: Please list the possible values. If this really is just 'enabled' and 'disabled', then change it to a boolean and convert it to the needed string (I assume for legacy) later on.

File: src/Form/Template.php [hunk 2]
```diff
@@ -22,26 +21,74 @@
...
+
+	/**
+	 * Donation introduction content position.
+	 *
+	 * @var string $donationIntroductionContentPosition
+	 */
+	public $donationIntroductionContentPosition = '';
```
Comment (JasonTheAdams) [inline src/Form/Template.php:84]: Please provide a description with what this does along with the possible values.

**Comment by ravinderk** [comment]
@JasonTheAdams This has been updated as per your suggestion. Let me know if i miss something.

File: src/Form/Template.php [hunk 3]
```diff
@@ -131,26 +163,53 @@ public function getFailedDonationMessage() {
...
+	}
+
+	/**
+	 * Get donation introduction text.
 	 *
 	 * @since 2.7.0
+	 * @return string
+	 */
+	public function getDonationIntroductionContent() {
+		return '';
```
Comment (JasonTheAdams) [inline src/Form/Template.php:200]: Let's return null to represent a non-value.

**Comment by JasonTheAdams** [comment]
I just noticed Travis CI is failing. Can you please look into this?

**Comment by ravinderk** [comment]
@JasonTheAdams Because it is failing only for only a single instance, I keep an eye on failing tests and resolve it if continue.

**Pull request merged**